test/functional/connection.test.js

```diff
@@ -273,4 +274,34 @@ describe('Connection', function () {
       done();
     }
   });
+
+  it(
+    'should be able to connect again after close',
+    withClient(function (client, done) {
+      expect(client.isConnected()).to.be.true;
+
+      const collection = () => client.db('testReconnect').collection('test');
```

Comment:
```suggestion
      const collection = client.db('testReconnect').collection('test');
```
This isn't actually just a style nit, we should expect saving off a `Collection` like this to work through `MongoClient` reconnects, so we actually want to make a single instance and use it twice.

Reply:
I agree that this should work. Unfortunately, if we keep the same collection instance, then the topology used during the second `collection.insertOne()` references the original topology (which was closed during `client.close()`), causing a `MongoError: Topology is closed` for this test case. 

I believe this means we can't keep the same instance of `Db` or `Collection` around to use again, unless we only store the topology in the `MongoClient`, and whenever we want access to it in  a `Db` or `Collection`, we have to ask to `MongoClient` for it.

Reply:
@HanaPearlman that's a really good point, and a change I think we should make. Can you make a new NODE ticket for only storing a reference to the parent `MongoClient` in `Db`/`Collection`?